test/integration/entry-integration.js

```diff
@@ -419,4 +419,124 @@ describe('Entry Api', () => {
       expect(response.items[0].sys.publishedAt).to.not.be.undefined
     })
   })
+
+  describe('write with xspace references', () => {
+    let xSpaceEnabledSpace
+    let xSpaceEnabledEnvironment
+    let xSpaceDisabledSpace
+    let xSpaceDisabledEnvironment
+    let contentTypeDisabledSpace
+    let contentTypeEnabledSpace
+    let contentTypeData = {
+      name: 'testCTXSpace',
+      fields: [
+        {
+          id: 'title',
+          name: 'Title',
+          type: 'Text',
+        },
+        {
+          id: 'multiRefXSpace',
+          name: 'multiRefXSpace',
+          type: 'Array',
+          items: {
+            type: 'ResourceLink',
+            validations: [],
+          },
+          allowedResources: [
+            {
+              type: 'Contentful:Entry',
+              source: 'crn:contentful:::content:spaces/6mqcevu5a50r',
+              contentTypes: ['testCtxSpace'],
+            },
+          ],
+        },
+      ],
+    }
+
+    let entryData = {
+      fields: {
+        title: { 'en-US': 'this is the title' },
+        multiRefXSpace: {
+          'en-US': [
+            {
+              sys: {
+                type: 'ResourceLink',
+                linkType: 'Contentful:Entry',
+                urn: 'crn:contentful:::content:spaces/6mqcevu5a50r/entries/4zimQzVMxDsSX607PCfo2u',
+              },
+            },
+          ],
+        },
+      },
+    }
+
+    before(async () => {
+      // default space has xspace enabled in it through the feature flags
+      xSpaceEnabledSpace = await getDefaultSpace()
+      xSpaceEnabledEnvironment = await xSpaceEnabledSpace.getEnvironment('master')
+      xSpaceDisabledSpace = await createTestSpace(initClient(), 'Entry')
+      xSpaceDisabledEnvironment = await createTestEnvironment(
+        xSpaceDisabledSpace,
+        'Testing Environment'
+      )
+      contentTypeDisabledSpace = await xSpaceDisabledEnvironment.createContentTypeWithId(
+        generateRandomId('test-content-type'),
+        contentTypeData
+      )
+      await contentTypeDisabledSpace.publish()
+
+      contentTypeEnabledSpace = await xSpaceEnabledEnvironment.createContentTypeWithId(
+        generateRandomId('test-content-type'),
+        contentTypeData
+      )
+      await contentTypeEnabledSpace.publish()
+    })
+
+    after(async () => {
+      await contentTypeDisabledSpace.unpublish().then((unpublishedContentType) =>
+        unpublishedContentType.delete().then(() => {
+          xSpaceDisabledSpace.delete()
+        })
+      )
```

Comment:
nitpick: we should probably stick to `async/await` instead of `then().then()`

Reply:
that object is actually "refreshed" every time you perform an action in it, so you could write it as

```
await contentTypeDisabledSpace.unpublish()
await contentTypeDisabledSpace.delete()
await xSpaceDisabledSpace.delete()
```

Reply:
I wrote it that way before but for some reason I wanted to change it. I'll go back to this.